clang/test/CodeGenSYCL/int_header1.cpp

```diff
@@ -4,14 +4,14 @@
 // CHECK:template <> struct KernelInfo<class KernelName> {
 // CHECK:template <> struct KernelInfo<::nm1::nm2::KernelName0> {
 // CHECK:template <> struct KernelInfo<::nm1::KernelName1> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName3< ::nm1::nm2::KernelName0>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName3< ::nm1::KernelName1>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName4< ::nm1::nm2::KernelName0>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName4< ::nm1::KernelName1>> {
+// CHECK:template <> struct KernelInfo<::nm1::KernelName3<::nm1::nm2::KernelName0>> {
```

Comment:
We would need a space between '<' and '::' to avoid trigraph issues.

Reply:
OK. I'll look into that but the current integration header doesn't have this space emitted consistently. I think I can fix that though.

Reply:
What C++ standard is SYCL now?  If it is C++17, trigraphs don't exist :)

Reply:
I'm not sure. @Fznamznon  would you know?

Reply:
Also, @premanandrao :  Don't all trigraphs start with a question mark?  For digraphs (of which <: is one), they have to be whole tokens (and the :: is parsed greedily in C++), so I don't think those are a problem.

Reply:
Re the digraphs, see the callout here: https://en.cppreference.com/w/cpp/language/operator_alternative
Since c++11:
When the parser encounters the character sequence <:: and the subsequent character is neither : nor >, the < is treated as a preprocessing token by itself and not as the first character of the alternative token <:. Thus std::vector<::std::string> won't be wrongly treated as std::vector[:std::string>.

Reply:
Sorry, digraphs is what I was thinking of.  If there isn't a parsing issue, I am fine with there not being an explicit space.   Thanks!

Reply:
Our SYCL implementation uses C++14 by default.

Reply:
@Fznamznon, https://github.com/intel/llvm/pull/1662 (not yet merged) switches the default to C++17.